### PR TITLE
PLT-5405

### PR DESCRIPTION
**Pull request by aman-quartic**
SDK updates for illuminator API of event frames, rules and their occurrences and rulebreaks.

### CODE REVIEW

**Review by scriperdjq — COMMENTED** [review]
testcases?

**Review by scriperdjq — DISMISSED** [review]

**Review by shubham-quartic — DISMISSED** [review]

File: quartic_sdk/core/entities/asset.py [hunk 1]
```diff
@@ -40,11 +40,21 @@ def get_tags(self, query_params={}):
...
 
-    def event_frames(self):
+    def event_frames(self, query_params={}):
         """
         The given method returns the list of event frames for the given asset
+        :param query_params: Dictionary of filter conditions
         """
-        raise NotImplementedError
+        from quartic_sdk.core.entity_helpers.entity_factory import EntityFactory
+        query_params["asset"] = self.id
```
Comment (mprasoonquartic) [inline quartic_sdk/core/entities/asset.py:49]: can you check once? the value of `query_params` will end up persisted, and thus any follow-up API call would probably end up passing this query_params as well.

File: quartic_sdk/core/entities/event_frame.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        :param start_time: (epoch) Start_time of event frame occurrence
+        :param stop_time: (epoch) Stop_time of event frame occurrence
+        :param query_params: Dictionary of filter conditions
+        """
+        from quartic_sdk.core.entity_helpers.entity_factory import EntityFactory
+
+        event_frame_occurrence_response = self.api_helper.call_api(
+            Constants.GET_EVENT_FRAME_OCCURRENCES,
+            Constants.API_GET,
+            path_params=[self.id, start_time, stop_time],
```
Comment (mprasoonquartic) [inline quartic_sdk/core/entities/event_frame.py:29]: you should add a check for `start_time` and `stop_time` in epoch always and not datetime object

File: quartic_sdk/core/entities/event_frame_occurrence.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    """
+    The given class refers to the event frame occurrence entity which is created based upon the
+    event frame occurrence object returned from the API
+    """
+
+    def __repr__(self):
+        """
+        Override the method to return the event frame occurrence name
+        """
+        return f"<{Constants.EVENT_FRAME_OCCURRENCE_ENTITY}: {self.name}>"
```
Comment (mprasoonquartic) [inline quartic_sdk/core/entities/event_frame_occurrence.py:15]: don't think event frame occurrencee has name. Can you recheck?

Comment (aman-quartic) [reply quartic_sdk/core/entities/event_frame_occurrence.py:15]: image

Here it is same as the event frame name

Comment (mprasoonquartic) [reply quartic_sdk/core/entities/event_frame_occurrence.py:15]: this is the name of event frame definition nd not the occurrence. upddate this to print id. also, override init to replace name with event_frame_name

Comment (aman-quartic) [reply quartic_sdk/core/entities/event_frame_occurrence.py:15]: Done

File: quartic_sdk/core/entities/event_frame.py (hunk 2 above)
Comment (aman-quartic) [reply quartic_sdk/core/entities/event_frame.py:29]: Done

**Review by scriperdjq — APPROVED** [review]

**Review by scriperdjq — APPROVED** [review]

**Review by mprasoonquartic — APPROVED** [review]

**Pull request merged**